libcontainer/process_linux.go

```diff
@@ -400,7 +400,7 @@ func (p *initProcess) start() (retErr error) {
 					}
 					// initProcessStartTime hasn't been set yet.
 					s.Pid = p.cmd.Process.Pid
-					s.Status = configs.Creating
```

Comment:
Maybe change `Status` type from `string` to `specs.ContainerState` instead of typecasting?

Reply:
I was looking at that; I _think_ that would be a breaking change though, WDYT?

Reply:
I don't know :( but these typecasts everywhere just look wrong

@tianon @cyphar PTAL ^^^

Reply:
It won't be a breaking change -- the state information is serialised to JSON and the JSON form will remain the same (`json.Marshal` doesn't care about type aliases like `type ContainerState string`).

Reply:
Yeah, (Go) is a bit hairy in that respect; https://play.golang.org/p/LGJ3vs3ncu0

```go
package main

type Foo string

type Foobar struct {
	Bla Foo
}

const somethingFoo Foo = "something"

func main() {
	// This works
	_ = Foobar{Bla: "something"}

	// This works
	_ = Foobar{Bla: somethingFoo}

	// This won't work
	something := "something"
	_ = Foobar{Bla: something}
}
```

Reply:
whoa! mind blown! @thaJeztah

Reply:
Yeah, I've hit this in the past as well (I think `Foobar{Bla: "something"}` working is actually a mistake on the Go language side of things, because it leads to more edge-cases than would be ideal).

However I think it's fine to make a change like that -- the Go files aren't really what is being versioned by `SemVer`, it's the specification. While we would ideally not break upgrades, I think doing a major version bump on `runtime-spec` because of a type alias would be a little bit of an over-reaction. :wink: